sdk/core/System.ClientModel/src/Convenience/ResultPageOfT.cs

```diff
@@ -0,0 +1,44 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.ClientModel.Primitives;
+using System.Collections.Generic;
+
+namespace System.ClientModel;
+
+/// <summary>
+/// Represents the subset (or page) of results contained in a single response
+/// from a cloud service returning a collection of results sequentially over
+/// one or more calls to the service (i.e. a paged collection).
+/// </summary>
+public class ResultPage<T> : ResultCollection<T>
+{
+    private readonly IEnumerable<T> _values;
+
+    /// <summary>
+    /// Create a new instance of <see cref="ResultPage{T}"/>.
+    /// </summary>
+    /// <param name="values">The values contained in <paramref name="response"/>.
+    /// </param>
+    /// <param name="continuationToken">The token that can be used to request
+    /// the next page of results from the service, or <c>null</c> if this page
+    /// holds the final subset of values.</param>
+    /// <param name="response">The <see cref="PipelineResponse"/> holding the
+    /// collection values returned by the service.</param>
+    public ResultPage(IEnumerable<T> values, string? continuationToken, PipelineResponse response)
```

Comment:
actually can we make the ctor private and expose a public Create static method. This will allow us to change the implementation from IEnumerable<T> if we want to in the future.

Reply:
Sure - will update

Reply:
Addressed.